frontend/src/services/fetch/fetchers/agenciesFetcher.ts

```diff
@@ -1,55 +1,41 @@
 "server only";
 
 import { ApiRequestError } from "src/errors";
-import {
-  fetchAgencies,
-  searchAgencies,
-} from "src/services/fetch/fetchers/fetchers";
+import { JSONRequestBody } from "src/services/fetch/fetcherHelpers";
+import { searchAgencies } from "src/services/fetch/fetchers/fetchers";
 import { RelevantAgencyRecord } from "src/types/search/searchFilterTypes";
 import { flattenAgencies } from "src/utils/search/filterUtils";
+import { getStatusValueForAgencySearch } from "src/utils/search/searchUtils";
 
-// would have called this getAgencies, but technically it's a POST
-export const obtainAgencies = async (): Promise<RelevantAgencyRecord[]> => {
-  const response = await fetchAgencies({
-    body: {
-      pagination: {
-        page_offset: 1,
-        page_size: 1500, // 969 agencies in prod as of 3/7/25
-        sort_order: [
-          {
-            order_by: "created_at",
-            sort_direction: "ascending",
-          },
-        ],
-      },
-      filters: { active: true },
-    },
-    nextOptions: {
-      revalidate: 604800,
+export const performAgencySearch = async ({
+  keyword,
+  selectedStatuses,
+}: {
+  keyword?: string;
+  selectedStatuses?: string[];
+} = {}): Promise<RelevantAgencyRecord[]> => {
+  const requestBody: JSONRequestBody = {
+    pagination: {
+      page_offset: 1,
+      page_size: 1500, // 969 agencies in prod as of 3/7/25
+      sort_order: [
+        {
+          order_by: "agency_code",
+          sort_direction: "ascending",
+        },
+      ],
     },
-  });
-  const { data } = (await response.json()) as { data: RelevantAgencyRecord[] };
-  return data;
-};
-
-export const performAgencySearch = async (
-  keyword?: string,
-): Promise<RelevantAgencyRecord[]> => {
-  const response = await searchAgencies({
-    body: {
-      pagination: {
-        page_offset: 1,
-        page_size: 1500, // 969 agencies in prod as of 3/7/25
-        sort_order: [
-          {
-            order_by: "agency_code",
-            sort_direction: "ascending",
-          },
-        ],
+    filters: {
+      opportunity_statuses: {
+        one_of: getStatusValueForAgencySearch(selectedStatuses),
       },
-      filters: { active: true },
-      query: keyword || "",
     },
+  };
+  if (keyword) {
+    requestBody.query = keyword;
```

Comment:
search fails with an empty keyword, but returns all results if keyword is omitted entirely